### PR TITLE
Fix incorrect upsert query for BigQuery sink

### PR DESCRIPTION
**Pull request by zoe-L**
Fixing bug https://cdap.atlassian.net/browse/PLUGIN-676 BigQuery sink generates incorrect query when using all the input fields in relationTableKey field

Remove `WHEN MATCHED THEN` clause from upsert query when all fields are used as keys.

### CODE REVIEW

**Comment by zoe-L** [comment]
Confirmed that the UPSERT query succeeds now in CDAP 6.8.0:
Screen Shot 2022-10-13 at 4 24 21 PM

Also, the e2e tests were broken upstream, not by this PR.

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java [hunk 1]
```diff
@@ -624,6 +630,10 @@ public static String generateUpdateUpsertQuery(Operation operation,
         return String.format(UPDATE_QUERY, destination, fieldsForUpdate, sourceTable, criteria);
       case UPSERT:
         String insertFields = "`" + String.join("`, `", tableFieldsList) + "`";
+        if (fieldsForUpdate.isEmpty()) {
```
Comment (fernst) [inline src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java:634]: Add a comment explaining why we return one or the other

**Review by fernst — COMMENTED** [review]
LGTM, but please add the comment I asked for

File: src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java (hunk 1 above)
Comment (zoe-L) [reply src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQuerySinkUtils.java:634]: Added comment

**Review by fernst — APPROVED** [review]

**Pull request merged**